ledger/src/shred/shred_code.rs

```diff
@@ -47,6 +47,13 @@ impl ShredCode {
         }
     }
 
+    pub(super) fn merkle_root(&self) -> Option<Hash> {
```

Comment:
Can we return `-> Result<Hash, Error>` to preserve the error?
`Self::Legacy(_)` case would then return `Err(Error::InvalidShredType)`.